src/test/java/data/TaskManagerTest.java

```diff
@@ -362,4 +363,89 @@ void getEventsForDate_invalidDate_returnsNoEvents() {
         // Assert
         assertTrue(eventsForDate.isEmpty());
     }
+
+    @Test
+    void markTaskAsCompleted_validIndex_marksTask() throws TaskManagerException {
+        // Arrange
+        LocalDate date = LocalDate.now();
+        addTask(date, "Complete project report", TaskType.TODO, 
+                new String[]{null}, new String[]{null});
+
+        // Act
+        taskManager.markTaskAsCompleted(date, 0);
+        Task completedTask = taskManager.getTasksForDate(date).get(0);
+
+        // Assert
+        assertTrue(completedTask.isCompleted(), "Task should be marked as completed.");
+    }
+
+    @Test
+    void markTaskAsCompleted_invalidIndex_throwsException() throws TaskManagerException {
+        // Arrange
+        LocalDate date = LocalDate.now();
+        addTask(date, "Write unit test", TaskType.TODO, 
+                new String[]{null}, new String[]{null});
+
+        // Act & Assert
+        IndexOutOfBoundsException exceptionThrown = assertThrows(
+                IndexOutOfBoundsException.class, 
+                () -> taskManager.markTaskAsCompleted(date, 1), 
+                "Should throw IndexOutOfBoundsException");
+
+        assertEquals(
```

Comment:
Testing both positive and negative test cases for each major function is incredibly comprehensive and will ensure we are more prepared to deal with bugs for the PE, so amazing job on that